src/DirectionProvider.jsx

```diff
@@ -12,11 +12,12 @@ import { DIRECTIONS, CHANNEL } from './constants';
 
 const propTypes = forbidExtraProps({
   children: PropTypes.node.isRequired,
-  direction: directionPropType.isRequired,
+  direction: directionPropType,
```

Comment:
i think i'd prefer using `explicitNull` from `airbnb-prop-types`, and leaving the isRequired, since `undefined` is likely a bug.

Reply:
Hm `undefined` isn't necessarily a bug. E.g. if we use this for an input field, it will frequently start out empty.

Reply:
Sure - but if you intend it, you could choose to explicitly send `null`, indicating that it's not a bug. Whereas if you omitted it (thus, `undefined`), it might indeed be a bug.

Reply:
Ah I see, good point! I like the outcome that we change getTextDirection to return `null` instead of `undefined`!